iceoryx_hoofs/source/posix_wrapper/posix_access_rights.cpp

```diff
@@ -147,10 +147,7 @@ PosixUser::groupVector_t PosixUser::getGroups() const noexcept
     }
 
     gid_t userDefaultGroup = getpwnamCall->value->pw_gid;
-
-    /// NOLINTJUSTIFICATION @todo iox-#1614 use upcoming cxx::array
-    /// NOLINTNEXTLINE(hicpp-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-    gid_t groups[MaxNumberOfGroups];
+    containers::UninitializedArray<gid_t, MaxNumberOfGroups, containers::ZeroedBuffer> groups;
```

Comment:
Again I am not sure we need zero initialization, but here i can think we do. It also depends on `gid_t`, as with zero initialization we set only the bytes to zero. This may or may not correspond to zeros for `gid_t` (it does for say integer types of course).

If in doubt we can leave it. Before each `gid_t` was default initialized, but what this means depends on `gid_t`.
We need to clarify what is required here.

Reply:
`gid_t` should be of type unsigned integer. `groups` is used to store the groups a user belongs to. Therefore, I think it's better if `groups` is zeroed instead of containing garbage. @elfenpiff What do you think?

Reply:
> I think it's better if groups is zeroed instead of containing garbage

No. Setting something to zero just in case is dangerous as it gives a false sense of correctness at the expense of performance. Either there is a need to initialize it here and then the question is to what value or there is no need to initialize it to anything specifically, and then it can be left uninitialized.

Whether there is a need to initialize it here depends on the logic and I do not know that from looking at this particular line without context. It is similarly hard to tell whether 0 would be the correct initialization.

Reply:
@FerdinandSpitzschnueffler in my opinion we should initialize this to an invalid `gid_t` value. Sadly 0 is the `gid_t` value of the `root` group and a logical mistake can be long hidden since we are operating on sane values.
An invalid `gid_t` value would be `uint32_t::MAX` or when the uninitialized array would initialize the whole memory with binary 1.

Yes this may, but only may, has a negative performance impact. But this is only when the object is created and this happens not on the hot path. So I would argue, from a safety point of view, that it makes sense to initialize this array to have at least defined behavior when a failure occurs.

But I admit performance is also important so @MatthiasKillat what do you think about a compromise like:
* In debug / developer mode this array is initialized to catch bugs early
* In release mode this array is not initialized

This could be our default policy for the uninitialized array.

Reply:
Would it fail currently if it was not initialized?